### DISCUSSION TITLE
About constant declaration

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/csharplang#6805] [Q&A]
```
public class Test
{
    public const MyEnum MyEnum = MyEnum.SomeName; // CS0110: The evaluation of the constant value for 'Test.MyEnum' involves a circular definition
    public const MyEnum MyEnum1 = MyEnum.SomeName; // ok
}

public enum MyEnum
{
    SomeName
}
```

I don't know why I couldn't declare ```const MyEnum MyEnum = MyEnum.SomeName;```?

![image](https://user-images.githubusercontent.com/117262133/207826776-85b6875a-b65e-40f7-bfa3-990a65f9ed35.png)

**Comment by bernd5** [comment]
`MyEnum` is interpreted as static/const field access (of the current type).

To access the type and not the field you can qualify it (or change the name). You can write:
```csharp
public class Test
{
    public const MyEnum MyEnum = global::MyEnum.SomeName; // CS0110: The evaluation of the constant value for 'Test.MyEnum' involves a circular definition
    public const MyEnum MyEnum1 = MyEnum.SomeName; // ok
}

public enum MyEnum
{
    SomeName
}
```

@CyrusNajmabadi: is this a bug? I'm wondering why it causes an error only while the constant is not defined...
